src/transforms/filter.jl

```diff
@@ -0,0 +1,42 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f, table)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter{F} <: Transform
+  pred::F
+end
+
+# Default filter to be used when no input function is given
+# Returns true if any element in the row evaluates to true
+Filter() = Filter(row -> any(row))
+
+isrevertible(::Type{Filter}) = true
+
+function apply(transform::Filter, input_table)
+  # Converting to Tables.rowtable to allow length function and indexing 
+  table = Tables.rowtable(input_table)
+
+  # Get indices of the desired rows
+  indices = [i for i in range(1,length=length(table)) if transform.pred(table[i])]
```

Comment:
Try to split your lines of code into shorter lines with intermediate variables for improved readability. There is too much going on in a single line, it also makes it harder to debug bugs in the future.

Reply:
Changing lines 24-25 to this: 
```{julia}
  l = length(table)
  f = transform.pred

  # Get indices of the desired rows
  indices = [i for i in range(1,length=l) if f(table[i])]
```